doc/CMakeLists.txt

```diff
@@ -4,10 +4,17 @@ if (NOT DOXYGEN)
   return ()
 endif ()
 
+find_package(PythonInterp
+  QUIET REQUIRED)
+find_program(SPHINX_EXECUTABLE
+  NAMES sphinx-build sphinx-build-3)
+
 add_custom_target(doc
-  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/build.py ${FMT_VERSION}
+  COMMAND ${CMAKE_COMMAND} -E env SPHINX_EXECUTABLE=${SPHINX_EXECUTABLE}
```

Comment:
Let's invoke `PYTHON_EXECUTABLE` directly, not through cmake and pass `SPHINX_EXECUTABLE` as an optional command-line argument.